core/src/main/java/hudson/model/ItemGroupMixIn.java

```diff
@@ -329,6 +338,38 @@ public synchronized TopLevelItem createProject( TopLevelItemDescriptor type, Str
     }
 
     /**
+     * Populate a {@link Categories} object from a specific {@link ItemGroup}.
+     *
+     * @return A object that represents a set of {@link Category}
+     */
+    public static Categories getCategories(Authentication a, ItemGroup c) {
```

Comment:
:ant: Does this actually need to be a public part of the API, or could it just be inlined in `View`?

Yes I originally suggested putting it in this class but that was under the assumption that it would be called from a web method in `Folder`, whereas it seems you are calling it from a web method in `View` which would automatically work the same way for folders as for Jenkins root.